src/query/storages/common/cache/src/providers/rocksdb_cache.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chrono::Utc;
+use common_exception::Result;
+use rocksdb::TransactionDBOptions;
+
+use super::rocksdb_disk_cache::i64_merge_operator;
+use super::rocksdb_disk_cache::KeyTimeValue;
+use super::rocksdb_disk_cache::DISK_SIZE_KEY;
+use super::rocksdb_disk_cache::KEY2TIME_COLUMN_PREFIX;
+use super::rocksdb_disk_cache::ROCKSDB_LRU_ACCESS_COUNT;
+use super::rocksdb_disk_cache::TIME2KEY_COLUMN_PREFIX;
+
+// map key -> value
+static KEY2VALUE_COLUMN_PREFIX: &str = "_key2value";
+
+pub struct RocksDbCache {
+    db: rocksdb::TransactionDB,
+    limit: i64,
+}
+
+impl RocksDbCache {
+    pub fn new(path: &str, limit: i64) -> Result<RocksDbCache> {
```

Comment:
If we remove the whole path, the rocksdb is work next time?

Reply:
yes, rocksdb will recreate the whole data dir.